src/main/scala/firrtl/transforms/Dedup.scala

```diff
@@ -74,7 +74,11 @@ class DedupModules extends Transform with DependencyAPIMigration with PreservesA
     val dedupMap = DedupModules.deduplicate(c, noDedups.toSet, annos, renameMap)
 
     // Use old module list to preserve ordering
-    val dedupedModules = c.modules.map(m => dedupMap(m.name)).distinct
+    // Lookup what a module deduped to, if it is a duplicate, remove it
+    val dedupedModules = c.modules.flatMap { m =>
+      val mx = dedupMap(m.name)
+      if (mx.name == m.name) Some(mx) else None
```

Comment:
This separate change also feels like it's more efficient than `distinct`, but did it make a measurable difference, too?